helper/chart_data.js

```diff
@@ -2263,3 +2271,303 @@ const createSolutionsArray = async function (data) {
         console.log(err);
     }
 }
+
+
+//Report for entity level report in assessments
+exports.entityLevelReportData = async function (data) {
+
+    return new Promise(async function (resolve, reject) {
+       
+        //group the data based on completed date   
+        let groupedSubmissionData = await groupArrayByGivenField(data, "completedDate");
+
+        let completedDateKeys = Object.keys(groupedSubmissionData);
+
+        let threshold = config.druid.no_of_assessment_submissions_threshold ? config.druid.no_of_assessment_submissions_threshold : default_no_of_assessment_submissions_threshold;
+
+        if(typeof threshold !== "number"){
+            throw new Error("no_of_assessment_submissions_threshold value should be integer");
+        }
+        
+        let dateArray = [];
+        if (completedDateKeys.length > threshold) {
+
+            dateArray.push(completedDateKeys[0]);
+
+            for (i = completedDateKeys.length - 1; i >= 0; i--) {
```

Comment:
@deepakr123 - The current code runs empty for loop cycles. Consider using this - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice

Reply:
resolved